docs/design/rubygems.md

```diff
@@ -0,0 +1,450 @@
+
+**BUNDLE_DEPLOYMENT**: Disallow changes to the Gemfile.
+When the Gemfile is changed, and the lockfile has not been updated, running Bundler commands will be blocked.
+
+**BUNDLE_FORCE_RUBY_PLATFORM**: Ignore the current machine's platform and install only ruby platform gems.
+As a result, gems with native extensions will be compiled from source.
+
+See bundle config [documentation](https://bundler.io/v2.5/man/bundle-config.1.html).
+
+---
+
+Note: _BUNDLE_FORCE_RUBY_PLATFORM_ check is done via gemlock-parser in cachito.
+
+_BUNDLE_DEPLOYMENT_ might be helpful when building an image.
+Using the `--local` flag with the `bundle install`, ensures that all dependencies are installed from the cache without accessing the internet.
+This is uncommon, so we would have to force users to use this flag.
+By setting the `BUNDLE_DEPLOYMENT` environment variable, users do not have use the `--local` flag.
+
+_When installing gems, Bundler will also "fetch" the gems from the cache and store them inside `vendor` directory._
+
+Here is more verbose explanation of the `--deployment` available in the `bundle install` command:
+
+```bash
+--deployment
+```
+
+In deployment mode, Bundler will 'roll-out' the bundle for production or CI use. Please check carefully if you want to have this option enabled in your development environment.
+This option is deprecated in favor of the deployment setting.
+
+There is also `--prefer-local` flag, which will prefer the local cache over the remote source, but it ain't working at all.
+Environment variable `BUNDLE_ALLOW_OFFLINE_INSTALL` is not working either with `bundle install` for some reason,
+which could be probably the most logical solution in this case.
+
+##### Copy
+
+Copy the local configuration file from the user repository to {output_dir} and set BUNDLE_APP_CONFIG to the new location.
+Then, just append all the environment variables needed to the "new" copy of the user configuration file.
+Bundler will rewrite previous values with the new ones when installing gems.
+
+#### Inject
+
+The other solution would be to inject the config file directly and rewrite the values.
+
+### Metadata
+
+### git repository URL
+
+- git repository URL is used in other package managers as well
+- no version information available
+- gems in the repository are path dependencies in the `Gemfile.lock` ?!
+
+### `{gem}.gemspec` file
+
+- the file is optional
+- complete metadata about the gem
+
+Gemfile must contain a _gemspec_ line, + the `{gem}.gemspec` file must be present in the repository.
+Bundle will add the gem as a path dependency to the `Gemfile.lock` file.
+This could be done via gemlock-parser by checking the path.
+
+```ruby
+source "https://rubygems.org"
+
+gemspec
+...
+```
+
+```Gemfile.lock
+...
+PATH
+ remote: .
+ specs:
+ tmp (0.1.2)
+...
+```
+
+### PURL
+
+Examples from [github.com/purl-spec](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#gem).
```

Comment:
Do think we can add PURL examples for each of the existing dependency types (or more if there are corner cases that are interesting to clarify)?